lib/gollum-lib/wiki.rb

```diff
@@ -510,10 +510,8 @@ def revert_page(page, sha1, sha2 = nil, commit = {})
         sha2   = nil
       end
 
-      patch     = full_reverse_diff_for(page, sha1, sha2)
       committer = Committer.new(self, commit)
-      parent    = committer.parents[0]
-      committer.options[:tree] = @repo.git.apply_patch(parent.sha, patch)
+      committer.options[:tree] = @repo.git.revert(page ? page.path : nil, sha1, sha2, committer.parents[0])
```

Comment:
Yes, but `revert_commit` just calls `revert_page` with `nil` as the `page` argument. If we're changing the API anyway we can try to make this less counterintuitive. What about these three methods:
* private: `revert(page = nil, sha1, sha2, commit = {})` (implements current logic of `revert_page`)
* public: `revert_page(page, sha1, sha2, commit = {})` (calls `revert` with a page argument)
* public: `revert_commit(sha1, sha2, commit = {})` (calls revert_commit)